tests/debugger/test_debugger_inproduct_enablement.py

```diff
@@ -0,0 +1,153 @@
+# Unless explicitly stated otherwise all files in this repository are licensed under the the Apache License Version 2.0.
+# This product includes software developed at Datadog (https://www.datadoghq.com/).
+# Copyright 2021 Datadog, Inc.
+
+import tests.debugger.utils as debugger
+from utils import features, scenarios
+from utils.tools import logger
+import json
+
+TIMEOUT = 5
+
+
+@features.debugger_inproduct_enablement
+@scenarios.debugger_inproduct_enablement
+class Test_Debugger_InProduct_Enablement_Dynamic_Instrumentation(debugger._Base_Debugger_Test):
+    ############ dynamic instrumentation ############
+    _probe_template = """
+    {
+        "version": 0,
+        "type": "LOG_PROBE",
+        "where": {
+            "typeName": null,
+            "sourceFile": "ACTUAL_SOURCE_FILE",
+            "lines": ["20"]
+        }
+    }
+    """
+
+    def setup_inproduct_enablement_di(self):
+        def _send_config(enabled=None):
+            probe = json.loads(self._probe_template)
+            probe["id"] = debugger.generate_probe_id("log")
+            self.set_probes([probe])
+
+            self.send_rc_apm_tracing(dynamic_instrumentation_enabled=enabled)
+
+            self.send_rc_probes()
+            self.send_weblog_request("/debugger/log")
+
+        self.initialize_weblog_remote_config()
+        self.weblog_responses = []
+        self.rc_states = []
+
+        _send_config()
+        self.di_initial_disabled = not self.wait_for_all_probes_emitting(TIMEOUT)
+
+        _send_config(enabled=True)
+        self.di_explicit_enabled = self.wait_for_all_probes_emitting(TIMEOUT)
+
+        _send_config()
+        self.di_empty_config = self.wait_for_all_probes_emitting(TIMEOUT)
+
+        _send_config(enabled=False)
+        self.di_explicit_disabled = not self.wait_for_all_probes_emitting(TIMEOUT)
```

Comment:
Are we waiting the full 5s to determine the probes are no longer emitting? Is there a more explicit response we can check for?

Reply:
Yes, we're waiting the full 5 seconds. We could probably change the behavior to check telemetry that should update when the product is enabled/disabled, but that's probably something else to add to the test, not something to rely on. 🤔